frontend/src/pages/QueryBuilder/QueryBuilder.tsx

```diff
@@ -43,19 +50,30 @@ interface ReduxState {
 interface RouterParams {
   id?: string;
 }
+interface QueryState {
+  previewData?: List<OperationDataMap>;
+  previewShow: boolean;
+  loadingPreview: boolean;
+}
 type QueryBuilderProps = ActionProps & ReduxState & RouteComponentProps<RouterParams>;
 
-const StyledIcon = styled.i`
-  cursor: pointer;
-`;
 const StyledCardBody = styled(Card.Body)`
   &.card-body {
     padding-right: 15px;
     padding-left: 15px;
   }
 `;
 
-class QueryBuilder extends React.Component<QueryBuilderProps> {
+class QueryBuilder extends React.Component<QueryBuilderProps, QueryState> {
```

Comment:
If you're going to be making such extensive changes to a component, you should first upgrade it from a class component to a function component.